### PR TITLE
Interface for Nar

### PR DESCRIPTION
**Pull request by PtrMan**

### CODE REVIEW

File: src/main/java/org/opennars/interfaces/HasPlugins.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.opennars.main.Nar;
+import org.opennars.plugin.Plugin;
+
+import java.util.List;
+
+/**
+ * Implementation can have plugins
+ */
+public interface HasPlugins {
```
Comment (freemo) [inline src/main/java/org/opennars/interfaces/HasPlugins.java]: I'd rename this to Pluggable

File: src/main/java/org/opennars/interfaces/HasSensoryChannels.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.opennars.interfaces;
+
+import org.opennars.plugin.perception.SensoryChannel;
+
+/**
+ * Implementations have sensory channels
+ */
+public interface HasSensoryChannels {
```
Comment (freemo) [inline src/main/java/org/opennars/interfaces/HasSensoryChannels.java]: nethods that start with has is resered in java for Boolean accessors. should rename this.Perhaps `SensoryChannelConsumer`

File: src/main/java/org/opennars/interfaces/StringNarsese.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.opennars.interfaces;
+
+/**
+ * Something which can work with Narsese as a string representation
+ */
+public interface StringNarsese {
```
Comment (freemo) [inline src/main/java/org/opennars/interfaces/StringNarsese.java]: Not a big deal wither way but doesnt NArsesString just sound better?

**Comment by PtrMan** [comment]
@TonyLo1 for all methods in the interfaces? I can give it a try :) Thank you for this feedback :)

**Comment by TonyLo1** [comment]
@PtrMan, yes please, all methods and params in all interfaces.

**Comment by PtrMan** [comment]
Can this pass review?

**Comment by PtrMan** [comment]
Thank you :)

**Pull request merged**